src/test/java/ch/uzh/ifi/hase/soprafs24/service/TaskServiceTest.java

```diff
@@ -356,17 +342,17 @@ public void apply_validInputs_applicationCreated() {
 
         User candidate = new User();
         candidate.setId(1L);
+        candidate.setApplications(new ArrayList<>(Collections.emptyList()));
 
         Task task = new Task();
         task.setId(1L);
 
-        when(userRepository.findUserByToken(anyString())).thenReturn(candidate);
-        when(taskRepository.findById(anyLong())).thenReturn(task);
-        when(applicationsRepository.findByUserAndTask(any(User.class), any(Task.class))).thenReturn(null);
 
-        taskService.apply(taskPutDTO, "validToken");
+        when(userService.getUserByToken(anyString())).thenReturn(candidate);
+        when(taskRepository.findById(anyLong())).thenReturn(Optional.of(task));
 
-        verify(applicationsRepository, times(1)).saveAndFlush(any(Application.class));
+        taskService.apply(taskPutDTO, "validToken");
+        assertTrue(task.getCandidates().contains(candidate));
```

Comment:
This replaces testing that the saveAndFlush has been called once, we check if the list of candidates for the task contains the newly candidate user.